src/Avalonia.Controls/TopLevel.cs

```diff
@@ -365,5 +368,17 @@ private void SceneInvalidated(object sender, SceneInvalidatedEventArgs e)
         {
             (this as IInputRoot).MouseDevice.SceneInvalidated(this, e.DirtyRect);
         }
+
+        void PlatformImpl_LostFocus()
+        {
+            var focused = (IVisual)FocusManager.Instance.Current;
+            if (focused == null)
+                return;
+            while (focused.VisualParent != null)
+                focused = focused.VisualParent;
+
+            if (focused == this)
+                KeyboardDevice.Instance.SetFocusedElement(null, NavigationMethod.Unspecified, InputModifiers.None);
```

Comment:
@grokys this code is moved from our winforms host where it was since forever. But I'm not sure if that's the correct way of informing our focus management system that the current window has lost focus and it's now held by a child window.

Reply:
Yeah our focus management has no concept of child windows so not sure we can do much better than this with the way things stand.